lib/saluki-components/src/sources/dogstatsd/mod.rs

```diff
@@ -425,6 +425,8 @@ async fn drive_stream(
                                             .set_origin_entity(OriginEntity::ProcessId(creds.pid as u32));
                                     }
                                 }
+                                Event::EventD(_) => todo!(),
+                                Event::ServiceCheck(_) => todo!(),
```

Comment:
Similar comment as the one for the Prometheus destination around just using `Event::into_metric` instead... although here we're only mutating the event, so we would need to add something like:


```rust
impl Event {
  pub fn as_metric_mut(&mut self) -> Option<&mut Metric> {
    match self {
      Self::Metric(mut metric) => Some(metric),
      _ => None,
    }
  }
}
```

(Code might not exactly compile but I think it captures the intent.)